torchgeo/datasets/utils.py

```diff
@@ -81,6 +80,26 @@ def __exit__(self, exc_type: None, exc_value: None, traceback: None) -> None:
             pass
 
 
+class _zipfile:
```

Comment:
If we didn't need test coverage, this entire class could be replaced by:
```python
try:
    import zipfile_deflate64 as zipfile
except ImportError:
    import zipfile
```
If zipfile_deflate64 is not installed and a user opens a deflate64-compressed zip file, they'll see the error message:
```
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
  File "/Users/Adam/.spack/.spack-env/._view/tmizjnr2yebouizawyxzmo3gwignw5lz/lib/python3.8/zipfile.py", line 1647, in extractall
    self._extract_member(zipinfo, path, pwd)
  File "/Users/Adam/.spack/.spack-env/._view/tmizjnr2yebouizawyxzmo3gwignw5lz/lib/python3.8/zipfile.py", line 1700, in _extract_member
    with self.open(member, pwd=pwd) as source, \
  File "/Users/Adam/.spack/.spack-env/._view/tmizjnr2yebouizawyxzmo3gwignw5lz/lib/python3.8/zipfile.py", line 1571, in open
    return ZipExtFile(zef_file, mode, zinfo, pwd, True)
  File "/Users/Adam/.spack/.spack-env/._view/tmizjnr2yebouizawyxzmo3gwignw5lz/lib/python3.8/zipfile.py", line 819, in __init__
    self._decompressor = _get_decompressor(self._compress_type)
  File "/Users/Adam/.spack/.spack-env/._view/tmizjnr2yebouizawyxzmo3gwignw5lz/lib/python3.8/zipfile.py", line 720, in _get_decompressor
    _check_compression(compress_type)
  File "/Users/Adam/.spack/.spack-env/._view/tmizjnr2yebouizawyxzmo3gwignw5lz/lib/python3.8/zipfile.py", line 700, in _check_compression
    raise NotImplementedError("That compression method is not supported")
NotImplementedError: That compression method is not supported
```
We could potentially catch this error and add a more helpful error message like we do with rarfile, but it isn't clear how to test this (see PR description).